packages/lib/modules/pool/migrations/MigrateLiquidityProvider.tsx

```diff
@@ -0,0 +1,73 @@
+import { useMandatoryContext } from '@repo/lib/shared/utils/contexts'
+import { createContext, PropsWithChildren, useEffect } from 'react'
+import { GetPoolDocument } from '@repo/lib/shared/services/api/generated/graphql'
+import { useQuery } from '@apollo/client'
+import { usePoolMigrations } from './PoolMigrationsProvider'
+import { getGqlChain } from '@repo/lib/config/app.config'
+import { useUserAccount } from '../../web3/UserAccountProvider'
+import { Address, zeroAddress } from 'viem'
+import { Pool } from '../pool.types'
+import { useRemoveLiquidity } from '../actions/remove-liquidity/RemoveLiquidityProvider'
+import { useAddLiquidity } from '../actions/add-liquidity/AddLiquidityProvider'
+import { useTransactionSteps } from '../../transactions/transaction-steps/useTransactionSteps'
+
+export type UseMigrateLiquidityResponse = ReturnType<typeof useMigrateLiquidityLogic>
+export const MigrateLiquidityContext = createContext<UseMigrateLiquidityResponse | null>(null)
+
+function useMigrateLiquidityLogic(protocol: number, chainId: number, poolId: string) {
+  const { userAddress } = useUserAccount()
+
+  const { transactionSteps: removeLiquiditySteps, amountsOut } = useRemoveLiquidity()
+  const { transactionSteps: addLiquiditySteps, setHumanAmountsIn } = useAddLiquidity()
+  const migrationSteps = useTransactionSteps(
+    [...removeLiquiditySteps.steps, ...addLiquiditySteps.steps],
+    removeLiquiditySteps.isLoading || addLiquiditySteps.isLoading
+  )
+
+  const { getMigration } = usePoolMigrations()
+  const migration = getMigration(protocol, chainId, poolId)
+
+  const { data: oldPoolData } = useFetchPool(chainId, poolId, userAddress)
+  const { data: newPoolData } = useFetchPool(migration?.new.chainId, migration?.new.id, userAddress)
+
+  useEffect(() => {
+    setHumanAmountsIn(amountsOut)
+  }, [amountsOut])
```

Comment:
how does this avoid upsetting linter about [set-state-in-effect](https://react.dev/reference/eslint-plugin-react-hooks/lints/set-state-in-effect)?

Reply:
probably because it is in other hook? (don't really know for sure)